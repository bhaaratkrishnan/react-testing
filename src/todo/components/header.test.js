import React from "react";
import { render } from "@testing-library/react";
import { Header } from "./header";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";

describe("Header", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Header />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
