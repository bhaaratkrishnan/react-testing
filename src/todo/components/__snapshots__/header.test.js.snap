// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
    data-testid="header"
  >
    <h1>
      todos
    </h1>
    <div
      class="input-container"
    >
      <input
        class="new-todo"
        data-testid="text-input"
        id="todo-input"
        placeholder="What needs to be done?"
        type="text"
        value=""
      />
      <label
        class="visually-hidden"
        for="todo-input"
      >
        New Todo Input
      </label>
    </div>
  </header>
</DocumentFragment>
`;
